### PR TITLE
Jussihakosalo/indica311

### PR DESCRIPTION
**Pull request by hakosaj**
Indica now compatible with 3.11.

### CODE REVIEW

File: indica/readers/abstractreader.py [hunk 1]
```diff
@@ -269,17 +268,19 @@ def get_charge_exchange(
         """
         Reads Charge-exchange-spectroscopy data
         """
+
         database_results = self._get_charge_exchange(
             uid, instrument, revision, quantities
         )
-
+        print("charge4")
```
Comment (michael-gemmell) [inline indica/readers/abstractreader.py:275]: What are these for?

File: indica/utilities.py [hunk 2]
```diff
@@ -320,14 +320,13 @@ def format_coord(data: LabeledArray, var_name: str):
...
-    coords = [(var_name, DataArray(data, dims=var_name))]
-    return format_dataarray(data, var_name, coords)
+    return format_dataarray(data, var_name, {var_name: data})
 
 
 def format_dataarray(
     data: LabeledArray,
     var_name: str,
-    coords: List[Tuple[str, Any]] = [],
+    coords: Dict[str, Any] = {},
```
Comment (michael-gemmell) [inline indica/utilities.py:329]: This should be = None right? Since it's mutable

**Review by michael-gemmell — COMMENTED** [review]
Looks good to me, just have to fix CI/CD.

File: indica/readers/abstractreader.py (hunk 1 above)
Comment (hakosaj) [reply indica/readers/abstractreader.py:275]: Just weird debugging I forgot to remove

**Review by marcosertoli — APPROVED** [review]
All good, just need to sort CI

**Comment by pixelifytica** [comment]
Why have we dropped support for python 3.9 with these changes?

**Comment by pixelifytica** [comment]
From what I can see the only incompatible package is `ipython`, everything else seems still work with 3.9 (at least they're supported versions, I'm just running some tests now to make sure nothing odd happens), so we're safe to keep 3.9 support without changing the supported versions of the other dependencies.

Do we need `ipython` as a dependency? I can't find it being used anywhere

**Comment by marcosertoli** [comment]
We don't need ipython, so it can be deleted.

I use it, but it's my personal development route, so no need to keep it in there.

Marco


________________________________

**Comment by pixelifytica** [comment]
Not a problem to move `ipython` to the dev dependencies group then rather than remove it, we can also set it to be conditional on python version so it's only installed if supported

**Pull request merged**